### PR TITLE
bpo-36409: Remove old plistlib API deprecated in 3.4

### PR DESCRIPTION
**Pull request by bigfootjon**
Also cleanup the documentation

https://bugs.python.org/issue36409

### CODE REVIEW

File: Doc/whatsnew/3.9.rst [hunk 1]
```diff
@@ -207,6 +207,11 @@ Removed
   Use :meth:`~threading.Thread.is_alive()` instead.
   (Contributed by Dong-hee Na in :issue:`37804`.)
 
+* The old :mod:`plistlib` API has been removed, it was deprecated since Python
+  3.4: use :func:`load`, :func:`loads`, :func:`dump`, :func:`dumps`, and
+  standard :class:`bytes` objects instead.
+  (Contributed by Jon Janzen)
+
```
Comment (ronaldoussoren) [inline Doc/whatsnew/3.9.rst:221]: Please also mention that the "use_builtin_types" parameter was removed.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]
The patch looks good to me, except for one minor issue.

File: Doc/whatsnew/3.9.rst (hunk 1 above)
Comment (ronaldoussoren) [reply Doc/whatsnew/3.9.rst:221]: Thanks.

File: Doc/whatsnew/3.9.rst [hunk 2]
```diff
@@ -213,6 +213,12 @@ Removed
   instead.  The ``xml.etree.cElementTree`` module has been removed.
   (Contributed by Serhiy Storchaka in :issue:`36543`.)
 
+* The old :mod:`plistlib` API has been removed, it was deprecated since Python
+  3.4. Use the :func:`load`, :func:`loads`, :func:`dump`, and :func:`dumps`
+  functions. Additionally, the `use_builtin_types` parameter was removed,
```
Comment (ronaldoussoren) [inline Doc/whatsnew/3.9.rst]: There is a CI failure due to the way `use_buitin_types` is written.  Please use double backticks.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]
Just one more nit to pick... 

I will merge once this is changed and the CI passes.

**Comment by Bigfootjon** [comment]
Thank you for the quick turnaround and tips @ronaldoussoren, I appreciate the help!

I have made the requested changes; please review again

**Review by ronaldoussoren — APPROVED** [review]
Patch looks good to me.

**Comment by ronaldoussoren** [comment]
@Bigfootjon Thanks for your patch!

**Comment by justvanrossum** [comment]
It appears the removal of the `use_builtin_types` argument to `load()` was not preceded by a deprecation phase. That makes this a rather hard breaking change. See for example github.com/unified-font-object/ufoNormalizer/issues/73 and downstream issues.

**Pull request merged**